consumer.go

```diff
@@ -231,6 +283,111 @@ func (c *Consumer) ScanShard(ctx context.Context, shardID string, fn ScanFunc) e
 	}
 }
 
+// ScanShardWithIntervalBatching waits for a specific second time interval to occur and then performs a batch fetch with specific shard, calls the callback func
+// with an array of records and checkpoints the scans
+
+func (c *Consumer) ScanShardWithIntervalBatching(ctx context.Context, shardID string, fn ScanFuncBatch) error {
+	// get last seq number from checkpoint
+
+	lastSeqNum, err := c.group.GetCheckpoint(c.streamName, shardID)
+	if err != nil {
+		return fmt.Errorf("get checkpoint error: %v", err)
+	}
+
+	// get shard iterator
+	shardIterator, err := c.getShardIterator(ctx, c.streamName, shardID, lastSeqNum)
+	if err != nil {
+		return fmt.Errorf("get shard iterator error: %v", err)
+	}
+
+	c.logger.Log("[CONSUMER] start scan:", shardID, lastSeqNum)
+	defer func() {
+		c.logger.Log("[CONSUMER] stop scan:", shardID)
+	}()
+	scanTicker := time.NewTicker(c.scanInterval)
+	defer scanTicker.Stop()
+	time_start := time.Now()
+	for {
+
+		if time_elapsed := time.Since(time_start); int64(time_elapsed.Seconds()) <= c.batchInterval {
```

Comment:
what is this doing?

Reply:
Effectively just a poorer duplication of scan interval, when I wrote this initially I overlooked that component of the code... thank you for pointing this out